src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1297,6 +1429,21 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::Get(spinel_prop_key_t aK
     va_end(mPropertyArgs);
     mPropertyFormat = nullptr;
 
+#if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_ENABLE
+    if (mRcpFailed)
+    {
+        RecoverFromRcpFailure();
+
+        mPropertyFormat = aFormat;
+        va_start(mPropertyArgs, aFormat);
+        error = RequestV(true, SPINEL_CMD_PROP_VALUE_GET, aKey, nullptr, mPropertyArgs);
+        va_end(mPropertyArgs);
+        mPropertyFormat = nullptr;
```

Comment:
```suggestion
        mPropertyFormat = aFormat;
        va_start(mPropertyArgs, aFormat);
        error = RequestV(true, SPINEL_CMD_PROP_VALUE_GET, aKey, nullptr, mPropertyArgs);
        va_end(mPropertyArgs);
        mPropertyFormat = nullptr;
```
Reuse the code before `if (mRcpFailed)` somehow to reduce code redudancy?

Reply:
There are some subtle differences in these codes, e.g. `RequestV` at <https://github.com/openthread/openthread/blob/7bdc83f8b3afe782dc01a231a51b79579a9aab11/src/lib/spinel/radio_spinel_impl.hpp#L1428> and `Request` at <https://github.com/openthread/openthread/blob/7bdc83f8b3afe782dc01a231a51b79579a9aab11/src/lib/spinel/radio_spinel_impl.hpp#L1464>, `mPropertyFormat` at <https://github.com/openthread/openthread/blob/7bdc83f8b3afe782dc01a231a51b79579a9aab11/src/lib/spinel/radio_spinel_impl.hpp#L1426> and `mExpectedCommand` at <https://github.com/openthread/openthread/blob/7bdc83f8b3afe782dc01a231a51b79579a9aab11/src/lib/spinel/radio_spinel_impl.hpp#L1493>. Also, it may lead to long function names, e.g. `RequestAndWaitWithPropertyFormatV`. Thoughts?

Reply:
2 instances of similar code don't need refactoring. But I can try to merge 4 instances from get/getwithparam into one function, and 6 instances from set/insert/remove into another function.

Reply:
Did the refactoring in <https://github.com/openthread/openthread/pull/5694/commits/95880327ca7f92595227079cac7caaed531c738a>. Looks slightly less redundant. I'm OK with both versions.

Reply:
Maybe we had some misunderstanding. I was actually talking about the same issue as discuessed in https://github.com/openthread/openthread/pull/5694#discussion_r528461928 that there are code redundancies in `Get*` methods. 

I am resolving this since that issue is resolved.